cli/cmd/install/install_test.go

```diff
@@ -0,0 +1,163 @@
+	require.Contains(t, err.Error(), "found PVCs from previous installations (default/consul-server-test1,default/consul-server-test2), delete before re-installing")
+
+	// Clear out the client and make sure the check now passes.
+	c.kubernetes = fake.NewSimpleClientset()
+	err = c.checkForPreviousPVCs()
+	require.NoError(t, err)
+
+	// Add a new irrelevant PVC and make sure the check continues to pass.
+	pvc = &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "irrelevant-pvc",
+		},
+	}
+	c.kubernetes.CoreV1().PersistentVolumeClaims("default").Create(context.TODO(), pvc, metav1.CreateOptions{})
+	err = c.checkForPreviousPVCs()
+	require.NoError(t, err)
+}
+
+func TestCheckForPreviousSecrets(t *testing.T) {
+	c := getInitializedCommand(t)
+	c.kubernetes = fake.NewSimpleClientset()
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-consul-bootstrap-acl-token",
+		},
+	}
+	c.kubernetes.CoreV1().Secrets("default").Create(context.TODO(), secret, metav1.CreateOptions{})
+	err := c.checkForPreviousSecrets()
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "found consul-acl-bootstrap-token secret from previous installations: \"test-consul-bootstrap-acl-token\" in namespace \"default\". To delete, run kubectl delete secret test-consul-bootstrap-acl-token --namespace default")
+
+	// Clear out the client and make sure the check now passes.
+	c.kubernetes = fake.NewSimpleClientset()
+	err = c.checkForPreviousSecrets()
+	require.NoError(t, err)
+
+	// Add a new irrelevant secret and make sure the check continues to pass.
+	secret = &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "irrelevant-secret",
+		},
+	}
+	c.kubernetes.CoreV1().Secrets("default").Create(context.TODO(), secret, metav1.CreateOptions{})
+	err = c.checkForPreviousSecrets()
+	require.NoError(t, err)
+}
+
+// TestValidateFlags tests the validate flags function.
+func TestValidateFlags(t *testing.T) {
+	// The following cases should all error, if they fail to this test fails.
+	testCases := []struct {
+		input       []string
+		description string
+	}{
+		{[]string{"foo", "-auto-approve"}, "Should disallow non-flag arguments."},
+		{[]string{"-f='f.txt'", "-preset=demo"}, "Should disallow specifying both values file AND presets."},
+		{[]string{"-preset=foo"}, "Should error on invalid presets."},
+		{[]string{"-namespace=\" preset\""}, "Should error on an invalid namespace. If this failed, TestValidLabel() probably did too."},
+		{[]string{"-f=\"does_not_exist.txt\""}, "Should have errored on a non-existant file."},
+	}
+
+	for _, testCase := range testCases {
+		c := getInitializedCommand(t)
+		t.Run(testCase.description, func(t *testing.T) {
+			if err := c.validateFlags(testCase.input); err == nil {
+				t.Errorf("Test case should have failed.")
+			}
+		})
+	}
+}
+
+// TestValidLabel calls validLabel() which checks strings match RFC 1123 label convention.
+func TestValidLabel(t *testing.T) {
+	testCases := []struct {
```

Comment:
similar change where `map[string]struct{}` could be used to make the description the key which should make the test slightly easier to read.

Reply:
made a change similar to the previous comment, kept the struct but changed the order and spacing for readability